knowledge-base/print-a-report-directly-at-client-side-without-displaying-it-in-a-viewer.md

```diff
@@ -41,15 +35,73 @@ In the above scenario, the PDF plugin is a must for the browser. The approach is
 
 To export the report in a PDF file, you can use a [ReportProcessor](/api/telerik.reporting.processing.reportprocessor) in code instead of a viewer.
 
+## Solution
+
+Here is sample approach you can use to achieve the above in an ASP.NET Core application.
+
+1. Creating a controller endpoint that returns the report exported in PDF format.
+
+	````C#
+public class HomeController : Controller
+	{
+		private IWebHostEnvironment _environment;
+	
+		public HomeController(IWebHostEnvironment environment)
+		{
+			_environment = environment;
+		}
+	
+		public IActionResult GenerateReportPDF(string reportName)
+		{
+			ReportProcessor reportProcessor = new ReportProcessor();
+			Telerik.Reporting.UriReportSource uriReportSource = new Telerik.Reporting.UriReportSource();
+			uriReportSource.Uri = Path.Combine(_environment.ContentRootPath, "Reports", reportName);
+			RenderingResult result = reportProcessor.RenderReport("PDF", uriReportSource, null);
+			
+			return File(result.DocumentBytes, result.MimeType);
+		}
+	}
+````
+
+2. Loading the generated PDF file within a hidden iframe and opening the browser's print dialog.
+
+	````JavaScript
+function printReport() {
+		fetch("/Home/GenerateReportPDF?reportName=Barcodes Report.trdp")
+			.then(res => {
+				if (res.status === 200) {
+					return res.blob();
+				} else {
+					console.log("Could not retrieve PDF document.");
+				}
+			})
+			.then(blob => {
+				let objectURL = URL.createObjectURL(blob);
+				
+				var iframe = document.createElement("iframe");
+				iframe.style.display = "none";
+				iframe.src = objectURL;
+				iframe.onload = function () {
+					iframe.contentWindow.print();
+				}
+				
+				document.body.appendChild(iframe);
+			});
+	}
+````
+
+You can download the [ASP.NET Core Client Print Example](https://github.com/telerik/reporting-samples/tree/master/PrintReportDirectlyAtClientSide) to see the full solution.
+
 ## Notes
 
-You can test the examples (ASP.NET WebForms and ASP.NET MVC) illustrating how to load a PDF file, generated by Telerik Reporting in code, in browser and rely on the browser's PDF plugin to print the document.
+You can also test the examples (ASP.NET WebForms and ASP.NET MVC) illustrating how to load a PDF file, generated by Telerik Reporting in code, in browser and rely on the browser's PDF plugin to print the document.
 
 The result PDF file rendered on the server is returned with **Content-Disposition** header set to **inline**.
 
  > The confirmation of the print operation cannot be avoided due to security restrictions.
 
 ## See Also
+
 [Printing Reports]({%slug telerikreporting/using-reports-in-applications/display-reports-in-applications/web-application/html5-report-viewer/printing-reports%})
 
 [MVC WebForms Client Print Example](resources/mvc-webforms-clientprint.zip)

```

Comment:
I suggest removing this and the VB print link below. They are actually examples of exporting the report to PDF and that is already covered in https://docs.telerik.com/reporting/knowledge-base/exporting-a-report-to-pdf-programmatically, add your sample link here instead(you can also keep it above)